### DISCUSSION TITLE
Lexical Documentation

### DISCUSSION THREAD

**Discussion by davbrito** [facebook/lexical#4144] [General]
I wanted to start a discussion regarding the documentation of the Lexical library. As you may know,  Lexical is undoubtedly one of the best options available for creating rich text editors using JavaScript.

However, as a developer who has been working with Lexical for some time, I have found that its lack of proper documentation can be a significant barrier to fully utilizing its capabilities. 

While I understand that this may take time and effort, it would be great if the Lexical team could consider giving priority to the development of comprehensive documentation for the library. This would undoubtedly help to ensure that more developers can take advantage of the flexibility, modularity, and customization that Lexical offers.

I believe that this effort would benefit not only the users of Lexical but also the project itself by increasing its visibility and popularity.

Thank you for your hard work and dedication to creating such a fantastic library. I look forward to hearing your thoughts on the matter.

**Comment by ZelCloud** [comment]
It's approaching close to 3 years since this post was made, and all of this is still a pain point of lexical today. Plugins like lexical-code have essentially no documentation how to set them up or use them. The advice given is look at lexical playground, but that's such a large elephant of code it's hard to find the pieces to pull out and take, plus it's intermixed with 2 versions. One of Prism and one of Shiki (which one should actually be used?). Not to mention some of the times I've checked the playground it was bugged or broken in one way or another. 

I don't think we need full documentation for everything and anything. Though would it be possible to have at least some of the most common cases well documented and maintained ?

Like for a start the following 2:
1. How to include and use one of the non trivial not text command plugins (ex. code plugin). 
2. How to create a custom plugin that isn't a simple replacement of the textnode.  Custom data going in and being displayed and exported.

**Reply by etrepum** [reply]
The best way to make it happen is to submit the PR yourself after you figure it out. Nobody is trying to sell lexical consulting so there isn’t really any commercial interest in docs and marketing, it’s all contributed.
